src/control/client/connect_test.go

```diff
@@ -292,10 +305,83 @@ func TestKillRank(t *testing.T) {
 
 	for _, tt := range tests {
 		cc := connectSetup(log, Ready, MockFeatures, MockCtrlrs, MockCtrlrResults, MockModules,
-			MockModuleResults, MockPmemDevices, MockMountResults, nil, nil, nil, nil, tt.killRet, nil)
+			MockModuleResults, MockPmemDevices, MockMountResults, nil, nil, nil,
+			nil, tt.killRet, nil, MockACL)
 
 		resultMap := cc.KillRank("acd", 0)
 
 		checkResults(t, Addresses{MockServers[0]}, resultMap, tt.killRet)
 	}
 }
+
+func TestPoolGetACL(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer ShowBufferOnFailure(t, buf)()
+
+	tests := []struct {
+		desc             string
+		addr             Addresses
+		getACLRespStatus int32
+		getACLErr        error
+		expectedACL      []string
+		expectedErr      string
+	}{
+		{
+			desc:        "no service leader",
+			addr:        nil,
+			expectedACL: nil,
+			expectedErr: "no active connections",
+		},
+		{
+			desc:        "gRPC call failed",
+			addr:        MockServers,
+			getACLErr:   MockErr,
+			expectedACL: nil,
+			expectedErr: MockErr.Error(),
+		},
+		{
+			desc:             "gRPC resp bad status",
+			addr:             MockServers,
+			getACLRespStatus: -5000,
+			expectedACL:      nil,
+			expectedErr:      "DAOS returned error code: -5000",
+		},
+		{
+			desc:             "success",
+			addr:             MockServers,
+			getACLRespStatus: 0,
+			expectedACL:      MockACL.acl,
+			expectedErr:      "",
+		},
+	}
+
+	for _, tt := range tests {
+		aclResult := &mockGetACLResult{
+			acl:    tt.expectedACL,
+			status: tt.getACLRespStatus,
+			err:    tt.getACLErr,
+		}
+		cc := connectSetupServers(tt.addr, log, Ready, MockFeatures,
+			MockCtrlrs, MockCtrlrResults, MockModules,
+			MockModuleResults, MockPmemDevices, MockMountResults,
+			nil, nil, nil, nil, nil, nil,
+			aclResult)
+
+		acl, err := cc.PoolGetACL("TestUUID")
+
+		if tt.expectedErr != "" {
+			ExpectError(t, err, tt.expectedErr, tt.desc)
+		} else if err != nil {
+			t.Fatalf("expected nil error, got %v", err)
+		}
+
+		if tt.expectedACL != nil {
```

Comment:
I think you could use cmp.Diff() to simplify this:
```go
  if diff := cmp.Diff(tt.expectedACL, acl); diff != "" {
      t.Fatalf("unexpected ACL (-want, +got):\n%s\n", diff)
  }
```

Reply:
Fixed